app/scripts/filters/navigate.js

```diff
@@ -0,0 +1,36 @@
+'use strict';
+
+angular.module('openshiftConsole')
+  // Resource is either a resource object, or a name.  If resource is a name, kind and namespace must be specified
+  // Note that builds and deployments can only have their URL built correctly (including their config in the URL)
+  // if resource is an object
+  .filter('navigateResourceURL', function(Navigate) {
+    return function(resource, kind, namespace, apiVersion) {
+      return Navigate.resourceURL(resource, kind, namespace, null, {apiVersion: apiVersion});
+    };
+  })
+  .filter('configURLForResource', function(Navigate) {
+    return function(resource, /* optional */ action) {
+      return Navigate.configURLForResource(resource, action);
+    };
+  })
+  .filter('editResourceURL', function(Navigate) {
+    return function(resource, kind, namespace) {
+      var url = Navigate.resourceURL(resource, kind, namespace, "edit");
+      return url;
+    };
+  })
+  .filter('editYamlURL', function(Navigate) {
+    return function(object, /* optional */ returnURL) {
+      return Navigate.yamlURL(object, returnURL);
+    };
+  })
+  .filter('urlForOwnerRef', function(Navigate) {
+    return function(ownerReference, namespace) {
+      if (!ownerReference) {
+        return null;
+      }
+
+      return Navigate.resourceURL(ownerReference.name, ownerReference.kind, namespace);
```

Comment:
If the ownerRef points to a resource that would navigate to the Other Resources page, and the user doesn't have List rights to that resource, I'm not sure what it will do.  Its not likely, but probably something we should at least handle be checking what kinds are available after the filtering when the page loads and showing some kind of generic "Could not list <resource>"

 Its an existing issue but this would make us more likely to hit it i think.